File: whereami.py
"""
ID: alexzha5
LANG: PYTHON3
TASK: whereami
"""
with open('whereami.in', 'r') as fin:
    num = int(fin.readline().strip())
    char = fin.readline().strip()

def f(s, n = 1):
    for i in range(num - n):
        if(s.count(s[i:i+n]) > 1):
            return f(s, n+1)
    return n

with open('whereami.out', 'w') as fout:
    fout.write(str(f(char)) + '\n')

File: whereami.cpp
/* Use the slash-star style comments or the system won't see your
   identification information */
/*
ID: alexzha5
TASK: whereami
LANG: C++
*/
/* LANG can be C++11 or C++14 for those more recent releases */
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    ofstream fout ("whereami.out");
    ifstream fin ("whereami.in");
    int n;
    string char;
    fin >> n >> char;

    fout << endl;
    return 0;
}
